libs/providers/flagd-web/src/lib/flagd-web-provider.ts

```diff
@@ -0,0 +1,240 @@
+import {
+  CallbackClient,
+  CallOptions,
+  Code,
+  ConnectError,
+  createCallbackClient,
+  createConnectTransport,
+  createPromiseClient,
+  PromiseClient
+} from '@bufbuild/connect-web';
+import { PartialMessage, Struct } from '@bufbuild/protobuf';
+import {
+  ErrorCode,
+  EvaluationContext, JsonValue,
+  Provider,
+  ProviderEvents,
+  ResolutionDetails,
+  StandardResolutionReasons
+} from '@openfeature/js-sdk';
+import { EventEmitter } from 'events';
+import NodeCache from 'node-cache';
```

Comment:
Probably shouldn't use this package for a web provider, it appears to [use NodeJS](https://github.com/node-cache/node-cache/blob/master/package.json#L62)

Reply:
Yeah it definitely uses NodeJS. It's built on the [Node.JS `events` module](https://nodejs.org/api/events.html).

Reply:
Ya, @beeme1mr brought this up too, it just has a nice API.

It definitely works in the web, but we might be risking some compatibility issues.

Reply:
I've implemented a `FlagdCache` interface, and an `LocalStorageFlagdCache` implementation. This is nice because it can be configured to persist data across page loads. What I've implemented isn't perfect (there's some key iteration that I wouldn't want to do in a very performance-constrained context) but for a web client with a number of flags in the order of 100s it should be sufficient for an experimental release.